utils/bookingRequest.ts

```diff
@@ -0,0 +1,127 @@
+
+export const defaultBookingDetailState: BookingDetailState = {
+  licensePlate: "",
+  checkInDate: null,
+  checkInTime: null,
+  checkOutDate: null,
+  checkOutTime: null,
+  specification: "None",
+  floor: -1,
+  slotId: "",
+  slotName: "",
+  price: -1,
+  unit: "",
+};
+
+export const createDefaultBookingDetailState = (
+  profile: Profile
+): BookingDetailState => {
+  const defaultLicensePlate = profile.user_cars
+    ?.filter((car) => car.is_default)
+    .map((defaultcar) => defaultcar._id)[0];
+  console.log(defaultLicensePlate);
+  return {
+    ...defaultBookingDetailState,
+    licensePlate: defaultLicensePlate ?? "",
+  };
+};
+
+export const validateTimeInputs = (bookingRequest: BookingDetailState) => {
+  const { checkInDate, checkInTime, checkOutTime, checkOutDate } =
+    bookingRequest;
+  if (checkInDate && checkInTime && checkOutTime && checkOutDate) {
+    if (
+      isCheckInTimeout(getDateFromDateAndTime(checkInDate, checkInTime)) ||
+      isCheckOutTimeout(getDateFromDateAndTime(checkOutDate, checkOutTime))
+    ) {
+      return ValidateStatus.TIMEOUT;
+    }
+    return ValidateStatus.SUCCESS;
+  }
+  return ValidateStatus.MISSING;
+};
+
+export const validateLicensePlate = (bookingRequest: BookingDetailState) => {
+  const { licensePlate } = bookingRequest;
+  return licensePlate != defaultBookingDetailState.licensePlate
+    ? ValidateStatus.SUCCESS
+    : ValidateStatus.MISSING;
+};
+
+export const validateAfterClosingSetting = (
+  bookingRequest: BookingDetailState
+) => {
+  const { slotId } = bookingRequest;
+  if (slotId == defaultBookingDetailState.slotId) {
+    return ValidateStatus.MISSING;
+  }
+  return ValidateStatus.SUCCESS;
+};
+
+export const getQueryParamFromBookingDetailState = (
+  bookingDetailState: BookingDetailState,
+  parkingLot: ParkingLot
+): GetAvailableSlotsQueryParam => {
+  const isValidTime =
+    validateTimeInputs(bookingDetailState) == ValidateStatus.SUCCESS;
+  const isValidLicensePlate =
+    validateLicensePlate(bookingDetailState) == ValidateStatus.SUCCESS;
+  // console.log("bookingDetailState", bookingDetailState);
+  if (isValidTime && isValidLicensePlate) {
+    return {
+      parkinglot_id: parkingLot._id,
+      start_date: bookingDetailState.checkInDate ?? "",
+      start_time: bookingDetailState.checkInTime
+        ? formatTimeWithSecond(bookingDetailState.checkInTime)
+        : "",
+      end_date: bookingDetailState.checkOutDate ?? "",
+      end_time: bookingDetailState.checkOutTime
+        ? formatTimeWithSecond(bookingDetailState.checkOutTime)
+        : "",
+      is_for_disabled: bookingDetailState.specification != "None",
+    };
+  }
+  return {
+    parkinglot_id: parkingLot._id,
+    start_date: null,
+    start_time: null,
+    end_date: null,
+    end_time: null,
+    is_for_disabled: false,
+  };
+};
+
+export const getTotalFloor = (slots: Slot[]) => {
+  return [...new Set(slots.map((item) => item.floor))];
+};
```

Comment:
This information should be provided from BE no?

Reply:
มีมาจากbackendแต่ยังไม่มีhierarchyจากbackendเลยทำท่านี้ไปก่อน สร้างการ์ดjiraทำhierarchyแล้ว